src/machines/modelingMachine.ts

```diff
@@ -3570,24 +3570,27 @@ export const modelingMachine = setup({
           return Promise.reject(new Error(NO_INPUT_PROVIDED_MESSAGE))
         }
 
-        const { target, tool } = input
+        const { solids, tools } = input
         if (
-          !target.graphSelections[0].artifact ||
-          !tool.graphSelections[0].artifact
+          !solids.graphSelections[0].artifact ||
+          !tools.graphSelections[0].artifact
         ) {
           return Promise.reject(new Error('No artifact in selections found'))
         }
 
-        await applySubtractFromTargetOperatorSelections(
-          target.graphSelections[0],
-          tool.graphSelections[0],
+        const result = await applySubtractFromTargetOperatorSelections(
+          solids,
+          tools,
           {
             kclManager,
             codeManager,
             engineCommandManager,
             editorManager,
           }
         )
+        if (err(result)) {
+          return Promise.reject(result)
+        }
```

Comment:
This bit was required for me to see error toasts coming out of this call, so I added it to all three bool ops for consitency